code/game/machinery/doors/door.dm

```diff
@@ -179,7 +179,7 @@
 
 /obj/machinery/door/take_damage(damage_amount, damage_type = BRUTE, damage_flag = 0, sound_effect = 1, attack_dir)
 	. = ..()
-	if(. && obj_integrity > 0)
+	if(. && obj_integrity > 0 && !QDELETED(src))

```

Comment:
take_damage should never be called after something is deleted so this should be unecessary